### PR TITLE
feat(integration): v0 integration suite gaps

### PR DESCRIPTION
**Pull request by char2cs**
## Summary

- **AsyncAPI spec** — adds `docs/asyncapi/asyncapi.yaml` (3.0) documenting the `/v0/arrow` and `/v0/runtime` WebSocket channels; wired into `make build-docs` for validation on every build
- **Integration test suite** — full coverage across lifecycle, guards, integrity, failures, edge cases, deps (cycles, self-ref, upgrade-cycle), concurrency, stress, crash recovery, oracle, reliability, and versioning scenarios; new fixtures and kit helpers for WS-based waiting, fetch serving, and state oracle
- **Runtime command split** — `BeginExecution` split into focused `BeginInstall`, `BeginUninstall`, `BeginStop`, `BeginUpdate` commands; idempotent install guard added at usecase layer; `_update` now works from both `Outdated` and `Ready` states; zero-step update completes cleanly
- **Benchmark suite** — `tests/integration/bench/` with 8 scenarios (Add, InstallCold, InstallWarm, GetDetail, List, DepResolution, StartupReplay, EventStoreDegradation); request→WS-event latency only, never polls; local baseline with `⚠ REGRESSION` warnings at 1.25× p99 threshold; gated behind `integration && bench` tags so `make test-integration` ignores it

## Commands

```bash
make test-integration   # run integration suite (bench excluded)
make bench-update       # first time: generate baseline for your machine
make bench              # check for regressions against baseline
```

## Test Plan

- [ ] `make test-integration` passes
- [ ] `make bench-update` generates `tests/integration/bench/baseline.json`
- [ ] `make bench` exits 0 (regressions warn but do not fail)
- [ ] `make build-docs` passes AsyncAPI validation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/app/repositories/runtime/internal/assembler/internal/steps.go
Comment (char2cs) [inline internal/app/repositories/runtime/internal/assembler/internal/steps.go:1]: What is this file doing? Is it really necessary?

File: internal/app/repositories/runtime/internal/commands/begin_stop.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	if current == nil || current.Ref == "" {
+		return fmt.Errorf("begin stop: %w", asynxModels.ErrValidation)
+	}
+	if current.Execution != nil && current.Execution.Method == domain.MethodStop {
+		return fmt.Errorf("begin stop: %w", asynxModels.ErrValidation)
+	}
+	switch current.State {
+	case domain.ArrowStateRunning, domain.ArrowStateDetached:
+		return nil
+	case domain.ArrowStateAbsent,
```
Comment (char2cs) [inline internal/app/repositories/runtime/internal/commands/begin_stop.go]: Couldn't we just use `default:`? It simpler than describing all possible states

File: internal/app/repositories/runtime/internal/assembler/internal/steps.go
Comment (char2cs) [reply internal/app/repositories/runtime/internal/assembler/internal/steps.go:1]: Yes, it's necessary — it's the single place that routes a method name to the correct lifecycle steps and AvailableIn states for the assembler. Extracted to its own file so it can be unit-tested in isolation (steps_test.go covers all method routing paths). Without it this logic would live inline inside the assembler.

File: internal/app/repositories/runtime/internal/commands/begin_stop.go (hunk 1 above)
Comment (char2cs) [reply internal/app/repositories/runtime/internal/commands/begin_stop.go]: Agreed — done. Switched both begin_install and begin_stop to `default: //nolint:exhaustive`. The exhaustive linter requires explicit cases on typed constants; the nolint directive lets us keep the simpler default form.
